packages/server/postgres/migrations/1627394777775_addPollsTable.ts

```diff
@@ -0,0 +1,58 @@
+import {MigrationBuilder, ColumnDefinitions} from 'node-pg-migrate'
+import {Client} from 'pg'
+
+import getPgConfig from '../getPgConfig'
+
+export const shorthands: ColumnDefinitions | undefined = undefined
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  const client = new Client(getPgConfig())
+  await client.connect()
+
+  await client.query(`
+    DO $$
+    BEGIN
+      IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'PollStatusEnum') THEN
+        CREATE TYPE "PollStatusEnum" AS ENUM (
+          'creating',
```

Comment:
i hadn't considered keeping the status on the poll in the DB!
thinking through it, I'm a little shy of putting editing state in our persisted DB because if i start to edit & then my browser crashes there's no TTL to reset the state. Years ago we did this with meeting facilitators disconnecting-- when the socket disconnected, we looked at all their meetings & reset the state. That's a lot of querying to do on every disconnect!

Here's an imperfect alternative that we could compare it with:
- replaced `status` with `endedAt` on the poll. a poll is either active or ended.
- refactor `EditCommentingMutation(isEditing, discussionId)` to something more generic like `EditDiscussion(isEditing, discussionId, editType: 'comment' | 'poll')`. 
- Georg's PR will persist that editing status in redis & we'll put a TTL on it to make sure nothing gets stuck
- in the clientSchema, we'll have a `newPoll: Poll` type on the discussion thread. then, when you click "create" it'll call `createPoll(newPoll)` that includes all the poll options

i may have forgotten some stuff. let me know your thoughts

Reply:
maybe instead of editType we have `threadableId`
then on the client we could determine the threadable type from the threadableId (store.get('threadableId').getType())
that way if someone is creating a new poll, we could just say `threableId: null` and that would mean "creating" instead of "editing"

Reply:
Yeah, good point! 
For now, I've added `endedAt` just to be able to _finish_ the poll. 

Until the user clicks `start` I _think_ everything should be local, e.g. poll should be added to the database only when a user added all the options and clicked `start`. When a user starts adding a poll it's best to use `EditCommentingMutation` and notify everybody that `user X creates a poll`. Same with editing.